### PR TITLE
initial commit on docker basic networking blogpost

### PR DESCRIPTION
**Pull request by basmoorkens**
pls review

### CODE REVIEW

File: _posts/2017-12-13-Docker-basic-networking.md [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+<p>
+    <img class="image" alt="Angular-backend call over external network" src="/img/docker-basic-networking/angular-by-external.png" />
+</p>
+
+The reason we cannot use the name of our backend REST container is pretty simple.
+
+AngularJS renders in the browser so it's not rendered inside a docker container.
+
+If you would use the name of the container, then our application would not know how to resolve that to an IP address as the name of the containers are only known within the Docker network.
+The reason it does work when you use the internal IP of the REST backend is that the IP address is known so the Nginx webserver in the Angular container knows how to route this to the REST backend.
```
Comment (TimDG) [inline _posts/2017-12-13-Docker-basic-networking.md]: Ben niet zeker wat ge hiermee bedoelt...

**Comment by yannickdeturck** [comment]
Don't forget to change the date in the filename before merging the PR. (`git mv ...`). :-)

File: _posts/2017-12-13-Docker-basic-networking.md [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+* Communication within a Docker network between containers
+* Communication outside the Docker network by exposing internal ports
+
+Our final application setup looks like this:  
+
+<p>
+    <img class="image" alt="Final docker setup" src="/img/docker-basic-networking/docker-basic-network-setup.png" />
+</p>
+
+All of our Docker application will be deployed on **one host machine**.
```
Comment (yannickdeturck) [inline _posts/2017-12-13-Docker-basic-networking.md]: application**s**

File: _posts/2017-12-13-Docker-basic-networking.md [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+We will have a custom Docker network running with **three containers** attached to that network:
+* The DB container is just a MySQL database running within Docker.
+* The Backend container is a Spring Boot application that connects to the MySQL DB container and provides a REST service to the outside world.
```
Comment (yannickdeturck) [inline _posts/2017-12-13-Docker-basic-networking.md]: backend

File: _posts/2017-12-13-Docker-basic-networking.md [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+* The Frontend container is an AngularJS application that consumes the REST service from the backend container.
```
Comment (yannickdeturck) [inline _posts/2017-12-13-Docker-basic-networking.md]: frontend

File: _posts/2017-12-13-Docker-basic-networking.md [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+    <img class="image" alt="Angular-backend call over external network" src="/img/docker-basic-networking/angular-by-external.png" />
+</p>
+
+The reason we cannot use the name of our backend REST container is pretty simple.
+
+AngularJS renders in the browser so it's not rendered inside a docker container.
+
+If you would use the name of the container, then our application would not know how to resolve that to an IP address as the name of the containers are only known within the Docker network.
+The reason it does work when you use the internal IP of the REST backend is that the IP address is known so the Nginx webserver in the Angular container knows how to route this to the REST backend.
+In this case, you have to use the internal port of REST backend as this request will travel over the docker network.
```
Comment (yannickdeturck) [inline _posts/2017-12-13-Docker-basic-networking.md]: Docker

File: _posts/2017-12-13-Docker-basic-networking.md [hunk 6]
```diff
@@ -0,0 +1,237 @@
...
+
+In the second case, the Nginx webserver in the Angular container communicates with our REST backend not through the Docker network.
+The reason that this works, is because we exposed the 8080 port of the REST backend to the outside world on the public accessible 8090 port.
```
Comment (yannickdeturck) [inline _posts/2017-12-13-Docker-basic-networking.md]: You can omit the comma here imo

Comment (yannickdeturck) [reply _posts/2017-12-13-Docker-basic-networking.md]: public -> publicly

**Comment by yannickdeturck** [comment]
Using the `docker-basic-networking.jpg` image as thumbnail is a bit blurry.
Try using resizing it for the thumbnail or picking a different image.
Size needs to be 500x293.

File: _posts/2017-12-13-Docker-basic-networking.md [hunk 7]
```diff
@@ -0,0 +1,237 @@
...
+`docker inspect mydb`
+
+<p>
+    <img class="image" alt="docker inspect mydb" src="/img/docker-basic-networking/mydb-inspect.png" />
+</p>
+
+We can see in the output that it has gotten the `172.18.0.2` IP address and that it's using the default gateway of the network that we created.
+Now we should set up the database in our container with the schema for our REST application.
+
+`MySQL -h 127.0.0.1 -P 3306  --user=root --password=test`
```
Comment (TimDG) [inline _posts/2017-12-13-Docker-basic-networking.md]: Is't commando niet `mysql` (all lowercase)?

Comment (basmoorkens) [reply _posts/2017-12-13-Docker-basic-networking.md]: klopt. ik weet niet vanwaar die bizare formatting komt

**Comment by yannickdeturck** [comment]
Resized higher quality sample of the thumbnail and renamed the file.
Good to go!

**Pull request merged**